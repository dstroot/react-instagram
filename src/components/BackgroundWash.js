import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const BackgroundWash = ({ filterOpts = [] }) => {

  // CREATES A FILTER OVER TOP
  const Filter = styled.div`
    position: fixed;
    width: 100vw;
    height: 100vh;
    background-image: linear-gradient(
      ${filterOpts.join()}
    );
    opacity: 0.7;
  `;

  return (
      <Filter />
  );
};

export default BackgroundWash;

BackgroundWash.propTypes = {
  /** Pass a single color or a gradient. Example of gradient: filterOpts={["to bottom right", "teal", "blue", "purple"]} */
  filterOpts: PropTypes.arrayOf(PropTypes.string),
};
