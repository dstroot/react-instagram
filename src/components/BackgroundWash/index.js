import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

export const BackgroundWash = ({ filterOpts = [] }) => {
  // creates a color "wash" over the page
  const Filter = styled.div`
    position: fixed;
    width: 100vw;
    height: 100vh;
    background-image: linear-gradient(${filterOpts.join()});
    opacity: 0.7;
  `;

  return <Filter />;
};

BackgroundWash.propTypes = {
  /** Pass a single color or a gradient. Example of gradient: filterOpts={["to bottom right", "teal", "blue", "purple"]} */
  filterOpts: PropTypes.arrayOf(PropTypes.string),
};
