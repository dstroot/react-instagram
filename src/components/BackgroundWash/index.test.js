// After importing the necessary packages, we can write a test which calls
// the render function provided by React Testing Library, which returns an
// object that we can extract asFragment from. Using this we can use the
// built-in functionality from Jest to perform a snapshot test.

import React from 'react';
import { render, cleanup } from '@testing-library/react';

// component to test
import BackgroundWash from '.';

// should be in setupTests.js?
afterEach(cleanup);

it('renders', () => {
  const { asFragment } = render(
    <BackgroundWash
      filterOpts={['to bottom right', 'teal', 'blue', 'purple']}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
