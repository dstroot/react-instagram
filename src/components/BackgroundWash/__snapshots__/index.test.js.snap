// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundWash renders 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa dnPAEz"
  />
</DocumentFragment>
`;
